service/frontend/workflowHandler.go

```diff
@@ -1783,9 +1780,21 @@ func (wh *WorkflowHandler) GetWorkflowExecutionRawHistory(
 		}
 
 		execution.RunId = common.StringPtr(token.RunID)
+
+		// we need to update the current next event ID
+		if len(token.PersistenceToken) == 0 {
+			token.BranchToken, _, nextEventID, err =
+				queryHistory(domainID, execution, token.BranchToken)
+			if err != nil {
+				return nil, wh.error(err, scope)
+			}
+			token.FirstEventID = token.NextEventID
+			token.NextEventID = nextEventID
+		}
```

Comment:
we are not going to do long poll, so why updating the first & next event id .& branch token?

Reply:
@anish531213 We can talk offline if you are not clear with this part.